### ISSUE TITLE
Add better ways to update the model parameters with new data (online learning)

### ISSUE THREAD

**Issue by ankurankan** [pgmpy/pgmpy#1765]
Some of the recent approaches are based on Expectation Maximization.

[1] Cohen, Ira, Alexandre Bronstein, and Fabio G. Cozman. "Online Learning of Bayesian Network Parameters." (2001).
[2] Liu, Jinzhong, and Qin Liao. "Online learning of Bayesian network parameters." 2008 Fourth International Conference on Natural Computation. Vol. 3. IEEE, 2008.

**Comment by Manas-7854** [comment]
Hi @ankurankan, i have gone through the given papers, can you tell me where exactly do i need to implement this?

**Comment by RudraCodesForU** [comment]
@ankurankan , Is it still open? Can I work on it

**Comment by ankurankan** [comment]
@RudraCodesForU Yes, but you will have to start with a bit of literature review to find the most suitable method to implement.

**Comment by RudraCodesForU** [comment]
@ankurankan , Sure I will look into the most suitable methods for model parameters updation.

**Comment by narutonamikaze** [comment]
Hey, i think The [DiscreteBayesianNetwork] class already has a [fit_update] method that supports online incremental parameter updating using Bayesian estimation with Dirichlet priors, leveraging previous CPDs and sample counts. 
However, to implement better ways to update model parameters with new data, we could : 
Expose more options for online learning in [fit_update] such as supporting Expectation Maximization (EM) for models with latent variables, or allowing the user to select the estimator and prior type.

Add documentation and examples to make these capabilities more discoverable.

Optionally, add a new method or argument to [fit_update] to support EM-based online updates.

**Comment by narutonamikaze** [comment]
Can you please assign this issue, and maybe i can try implementing some of the suggested methods . 
@ankurankan

**Comment by RudraCodesForU** [comment]
@narutonamikaze , I am already working on it, soon be making a PR regarding this issue. But your idea sparks me, if you have no problem we can discuss it further.

**Comment by ankurankan** [comment]
@RudraCodesForU @narutonamikaze Before starting to work on anything, I suggest sharing the methods that you found here. I think we will need to discuss it, both what methods would be interesting and the best ways to implement them.
